### PR TITLE
Set `InteractCheckableTool._roi` to `None` on `deactivate`

### PR DESCRIPTION
**Pull request by dhomeier**
## Description
Reset any `_roi` in an `InteractCheckableTool` when calling `deactivate`.
This is to avoid inheriting changed attributes like rotation angle from previously created selections. It will not reset the attributes when creating new ROIs with the same (left open) tool instance.

### CODE REVIEW

**Comment by pllim** [comment]
Is this trying to solve the same problem that I approached differently in https://github.com/glue-viz/glue-jupyter/pull/416 ? Or is this for something else?

**Comment by rosteen** [comment]
> Is this trying to solve the same problem that I approached differently in #416 ? Or is this for something else?

This is trying to solve the problem I was hacking around in https://github.com/spacetelescope/jdaviz/pull/2639.

**Comment by rosteen** [comment]
@dhomeier This looks good to go in to me, I confirmed on the Jdaviz side that it fixes the unwanted behavior.

**Comment by pllim** [comment]
tl;dr -- This does not reintroduce the problem where the angle resets when you click and drag a rotated shape, does it?

**Comment by rosteen** [comment]
> tl;dr -- This does not reintroduce the problem where the angle resets when you click and drag a rotated shape, does it?

The existing problem of the gray preview not matching the rotation of the actual subset persists, but it's not worse than the current state. The dragged subset retains its angle in my testing.

**Review by astrofrog — APPROVED** [review]
Looks good, thanks!
